src/mpid/ch4/netmod/ofi/ofi_am_impl.h

```diff
@@ -312,6 +312,11 @@ static inline int MPIDI_OFI_am_isend_short(int rank,
     iov[2].iov_base = (void *) data;
     iov[2].iov_len = data_sz;
 
+    if (data_sz & 0x7) {
```

Comment:
Couldn't this cause an out-of-bounds access at the sender? I think this issue should really be fixed in libfabric, if I understand it correctly.

Reply:
Yeah, wonder why the address sanitizer didn't trigger -- maybe because it is accessed by the sockets driver, which was not instrumented. 

How about we add padding as a fourth iov?

>  I think this issue should really be fixed in libfabric, if I understand it correctly.

I agree. But libfabric won't know what alignment we need. It should have an alignment setting that we can control.

Alternatively, we could disable `FI_MULTI_RECV`.

Reply:
> Yeah, wonder why the address sanitizer didn't trigger

Wait, this change is for UndefinedBehaviorSanitizer, which is made after I cleared asan tests. Maybe it will trigger now...

Reply:
> I agree. But libfabric won't know what alignment we need. It should have an alignment setting that we can control.

Control would be nice to have, but defaulting to malloc-like alignment for FI_MULTI_RECV (i.e. suitably aligned for any object type) would make this a whole lot simpler for the user. It might also be good to run ubsan with libfabric+fabtests to see if the same warning can be triggered without MPICH. That would be good to have if we decide to raise the issue in their Github.

Reply:
Fixing on the libfabric side is complicated. For one, it needs maintain both actual data_len and padded data_len, while not difficult but quite involved. For two, we have to fix for each provider.

On the other hand, the fix on mpich's side is rather trivial -- just pad the data. Now I used a static padding char array as an additional iov entry, the concern of accessing out-of-bound is removed.